### PR TITLE
Create `tower-test` and include tower-mock.

### PR DESCRIPTION
**Pull request by carllerche**
tower-mock is deleted in favor of having a single test crate. Once `tokio-test` is released, I would expect that it will be re-exported as part of `tower-test`.

Some changes are made to mock:

- `mock::pair()` is used to get the two handles and `Mock::new` is deleted.
- `Respond` is renamed to `SendResponse`.
- The fns are renamed `send_response` and `send_error`.
- The `Request` struct is removed in favor of just using a tuple.

This crate also includes a new macro: `assert_request_eq!` for working with the mock handle:

```rust
assert_request_eq!(mock_handle, "hello").send_response("world");
```

Not all the crates have been updated yet.

### CODE REVIEW

File: tower-test/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+//! Mock `Service` that can be used in tests.
+
+extern crate futures;
+extern crate tokio_sync;
+extern crate tower_service;
+
+mod macros;
```
Comment (LucioFranco) [inline tower-test/src/lib.rs]: does this support getting the macro via an import?

Comment (carllerche) [reply tower-test/src/lib.rs]: Not sure... I would guess so? It would be worth a test. We also probably should update everything to 2018...

Comment (carllerche) [reply tower-test/src/lib.rs]: If it is not, we can update it with #238.

File: tower-test/src/macros.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#[macro_export]
+macro_rules! assert_request_eq {
```
Comment (LucioFranco) [inline tower-test/src/macros.rs]: might want docs here?

Comment (carllerche) [reply tower-test/src/macros.rs]: Yeah. Probably... thanks for calling that out.

**Pull request merged**